main.py

```diff
@@ -8,12 +8,24 @@
 app = Flask(__name__)
 app.secret_key = SECRET_KEY
 
+# @app.route("/")
+# def home_page():
+#     if not 'hits' in session:
+#         session['hits'] = 0
+#     session['hits'] += 1
+#     return render_template("home_page.html", name="App Engine", hits=session['hits'])
+
 @app.route("/")
-def home_page():
-    if not 'hits' in session:
-        session['hits'] = 0
-    session['hits'] += 1
-    return render_template("home_page.html", name="App Engine", hits=session['hits'])
+def index():
+    return render_template("index.html")
+
+@app.route("/show/")
+def show():
+    return render_template("listing_show.html")
+
+@app.route("/form/")
+def form():
+    return render_template("listing_form.html")
```

Comment:
On creation, how about we create a new listing in "draft" mode, then redirect to the edit form?